resources/test262-agent-harness.js

```diff
@@ -0,0 +1,335 @@
+// Helper file to run a test262 within an agent.
+//
+// The target test262 is in text format. The harnessing code composes an HTML
+// page containing the test262, runs it and reports for errors if there was
+// any.  Similarly if the target agent is a Worker, the harnessing code embedes
+// the test code within the target worker, runs the test and checks out if
+// there were any errors.
+//
+// Currently supported agents are IFrame, Window, Worker and SharedWorker.
+
+// IFrame.
+function run_in_iframe_strict(test262, attrs, t) {
+    let opts = {}
+    opts.strict = true;
+    run_in_iframe(test262, attrs, t, opts);
+}
+
+function run_in_iframe(test262, attrs, t, opts) {
+  opts = opts || {};
+  // Rethrow error from iframe.
+  window.addEventListener('message', t.step_func(function(e) {
+    if (e.data[0] == 'error') {
+      throw new Error(e.data[1]);
+    }
+  }));
+  let iframe = document.createElement('iframe');
+  iframe.style = 'display: none';
+  content = test262_as_html(test262, attrs, opts.strict);
+  let blob = new Blob([content], {type: 'text/html'});
+  iframe.src = URL.createObjectURL(blob);
+  document.body.appendChild(iframe);
+
+  let w = iframe.contentWindow;
+  // Finish test on completed event.
+  w.addEventListener('completed', t.step_func(function(e) {
+    t.done();
+  }));
+  // If test failed, rethrow error.
+  let FAILED = 'iframe-failed' + opts.strict ? 'strict' : '';
+  window.addEventListener(FAILED, t.step_func(function(e) {
+    t.set_status(t.FAIL);
+    throw new Error(e.detail);
+  }));
+  // In case of error send it to parent window.
+  w.addEventListener('error', function(e) { 
+    e.preventDefault();
+    // If the test failed due to a SyntaxError but phase was 'early', then the
+    // test should actually pass.
+    if (e.message.startsWith("SyntaxError")) {
+        if (attrs.type == 'SyntaxError' && attrs.phase == 'early') {
+            t.done();
+            return;
+        }
+    }
+    top.dispatchEvent(new CustomEvent(FAILED, {detail: e.message}));
+  });
+}
+
+let HEADER = `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title></title>
+
+  <!-- Test262 harness -->
+  <script src="http://localhost:8000/resources/test262-harness.js"><\/script>
+
+  <!-- Test262 required libraries -->
+  <script src="http://localhost:8000/resources/test262/harness/assert.js"><\/script>
+  <script src="http://localhost:8000/resources/test262/harness/sta.js"><\/script>
```

Comment:
This domain doesn't work if the tests are run from `http://web-platform.test:8000` (in fact, my localhost:8000 is not the same as http://web-platform.test:8000)

Reply:
Add `.sub.` to the filename and then you can use substitutions like `{{host}}` c.f. the rather outdated documentation at https://wptserve.readthedocs.io/en/latest/pipes.html#sub

Reply:
(or use just paths, of course, in this case, but in general the feature is there if you need it).

Reply:
@jgraham to clarify, do you mean, eg:

```html
<script src="http://{{host}}:{{ports[http][0]}}/resources/test262/harness/sta.js"><\/script>
```

Reply:
Thanks for the hint. I wasn't happy with hardcoding the host and the port but I couldn't find other way to make it work. I'll fix.